eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -740,53 +750,46 @@ runImpRule stsEnv stsState stsSignal = do
 passTick ::
   forall era.
   ( HasCallStack
-  , BaseM (EraRule "TICK" era) ~ ReaderT Globals Identity
-  , Environment (EraRule "TICK" era) ~ ()
-  , STS (EraRule "TICK" era)
-  , Signal (EraRule "TICK" era) ~ SlotNo
-  , State (EraRule "TICK" era) ~ NewEpochState era
   , NFData (State (EraRule "TICK" era))
+  , ShelleyEraImp era
   ) =>
   ImpTestM era ()
 passTick = do
   impLastTick <- gets impLastTick
   curNES <- getsNES id
+  preAdaPots <- getAdaPots
   nes <- runImpRule @"TICK" () curNES impLastTick
   impLastTickL += 1
   impNESL .= nes
+  impAnn "Checking ADA preservation at the epoch boundary" $ do
```

Comment:
It might be an overkill to do this check for every TICK, since nothing changes except the pulser state on every tick. It is a valuable check on every `EPOCH` rule though, do you think it would make sense to move it to `passEpoch` instead?